### PR TITLE
Added a typealias subsection and regroup other type related subsectio…

### PR DESCRIPTION
**Pull request by paulmiard**
…ns into the existing 'Types' section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +16,6 @@
 	* [Error Handling](#error-handling)
 	* [Access Modifiers](#access-modifiers)
 	* [Imports](#imports)
-	* [Type Declarations](#type-declarations)
-	* [Type Inference](#type-inference)
```
Comment (Haud) [inline README.md:20]: why did you delete these?

File: README.md [hunk 2]
```diff
@@ -150,10 +148,68 @@ let scaleString = scale.stringValue
...
+Typealias declaration should precede any other type declaration.
+
+
+```swift
+
+typealias ClosureType = (ParameterTypes) -> (ReturnType)
+typealias AgeAndNameProcessor = (Integer, String) -> Void
+
+var intValue: Int
+
```
Comment (ThibaultKlein) [inline README.md:198]: you can remove these spaces

File: README.md [hunk 3]
```diff
@@ -150,10 +148,68 @@ let scaleString = scale.stringValue
...
+Typealias declaration should precede any other type declaration.
+
+
+```swift
+
+typealias ClosureType = (ParameterTypes) -> (ReturnType)
+typealias AgeAndNameProcessor = (Integer, String) -> Void
+
+var intValue: Int
```
Comment (ThibaultKlein) [inline README.md:197]: I would put a class or a struct declaration as an example instead, so it's clear that `typealias` is outside of a type scope

File: README.md (hunk 1 above)
Comment (ThibaultKlein) [inline README.md:20]: because everything is under `Type` now

Comment (paulmiard) [inline README.md:20]: Yeah they are now subsections of type so I figured they'd be better off the toc, otherwise we need one more tab

File: README.md (hunk 2 above)
Comment (paulmiard) [inline README.md:198]: done

File: README.md (hunk 3 above)
Comment (paulmiard) [inline README.md:197]: done. Also commented the first typealias example as it would not compile. Keeping it as a reference

**Comment by ThibaultKlein** [comment]
👍

**Comment by Haud** [comment]
What about typealias as part of protocol conformance? then it has to be part of the class

File: README.md [hunk 4]
```diff
@@ -150,10 +148,73 @@ let scaleString = scale.stringValue
...
+
+class Object {
+	private var someString = ""
+
+	func returnOne() {
+		return 1
+	}
+}
+```
+
```
Comment (paulmiard) [inline README.md:243]: re:Chris comment: `If declaring a typealias for protocol conformance, it should be declared at the top of the class, before anything else`

Comment (paulmiard) [inline README.md:243]: Done

File: README.md [hunk 5]
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
...
+In all use-cases, the colon should be associated with the left-most item with no spaces preceding and one space afterwards:
+
+```swift
+let myDictionary: [String: AnyObject] = ["String": 0]
+```
+
+#### typealias ####
+
+Typealias declaration should precede any other type declaration.
```
Comment (Haud) [inline README.md]: `Typealias declarations`

**Comment by Haud** [comment]
Small grammatical note, but otherwise 👍  for me once resolved

**Comment by paulmiard** [comment]
@ThibaultKlein I double checked it looks good to merge. I let you check as well if you want, or just click the merge button.

File: README.md [hunk 6]
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
 let scaleInt = scale.integerValue
 ```
 
+#### Type Declarations ####
+
+When declaring types, the colon should be placed immediately after the identifier followed by one space
+and the type name.
```
Comment (ThibaultKlein) [inline README.md]: remove the new line here

File: README.md [hunk 7]
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
...
+	}
+}
+```
+
+If declaring a typealias for protocol conformance, it should be declared at the top of the type declaration, before anything else.
+
+
+```swift
+protocol Configurable {
+    associatedtype InputData
```
Comment (ThibaultKlein) [inline README.md:216]: add a new line after `InputData`

**Comment by ThibaultKlein** [comment]
All your examples are missing the space between the first and last curly brackets (see Bracket Syntax section)

File: README.md [hunk 8]
```diff
@@ -150,10 +148,101 @@ let scaleString = scale.stringValue
...
+
+// Ok since the inferred type is not what we wanted:
+
+var max: Hashable = 0 // Compiler would infer Int, but we only want it to be hashable
+var name: String? = "John" // Compiler would infer this not to be optional, but we may need to nil it out later.
+```
+
+*Rationale* The compiler is pretty smart, and we should utilize it where necessary. It is generally obvious what the
+type is going to be in the instances above, so unless we need to be more explicit (as in the last examples above),
+it is better to omit unneeded words.
```
Comment (ThibaultKlein) [inline README.md]: remove the new line

File: README.md [hunk 9]
```diff
@@ -422,7 +490,7 @@ of how they wish to handle the result of this failable operation.
 
 #### NSError ####
 
-In general, you should avoid `NSError` in Swift in favor of defining your own `ErrorType`. However, in the event you do need to use `NSError` 
+In general, you should avoid `NSError` in Swift in favor of defining your own `ErrorType`. However, in the event you do need to use `NSError`
 (for interop with Objective-C, for instance):
```
Comment (ThibaultKlein) [inline README.md]: remove new line

File: README.md (hunk 6 above)
Comment (paulmiard) [inline README.md]: done (even though it doesn't matter it's markdown)

File: README.md (hunk 7 above)
Comment (paulmiard) [inline README.md:216]: done

File: README.md (hunk 8 above)
Comment (paulmiard) [inline README.md]: done (even though it doesn't matter it's markdown)

File: README.md (hunk 9 above)
Comment (paulmiard) [inline README.md]: done (even though it doesn't matter it's markdown)

**Comment by paulmiard** [comment]
@ThibaultKlein I addressed your comments then rebased and fixed conflicts. Let me know. If it looks good to you please merge. 

Also Atom removed a few extra spaces here and there for us..
